pkg/agent/manager/cache/cache.go

```diff
@@ -355,11 +360,39 @@ func (c *Cache) UpdateSVIDs(update *UpdateSVIDs) {
 			telemetry.SPIFFEID: record.entry.SpiffeId,
 		})
 		log.Debug("SVID updated")
+
+		// Registration entry is updated, remove it from stale map
+		delete(c.staleEntries, entryID)
 	}
 
 	c.notifyBySelectors(notifySet)
 }
 
+// GetStaleEntries obtains a list of stale entries
+func (c *Cache) GetStaleEntries() map[*common.RegistrationEntry]time.Time {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	staleEntries := make(map[*common.RegistrationEntry]time.Time)
+	for entryID := range c.staleEntries {
+		cachedEntry, ok := c.records[entryID]
+		if !ok {
+			c.log.WithField(telemetry.RegistrationID, entryID).Warn("stale entry not found")
+			// TODO: may we remove entry from stale?
+			delete(c.staleEntries, entryID)
```

Comment:
We have a read lock. Do we need a write lock?

Reply:
It still looks like we're doing this over a read lock.

Who calls this function and UpdateSVIDs? Are we already holding locks at that time?

Reply:
Both are call on sync https://github.com/MarcosDY/spire/blob/refactor-sycn-updates/pkg/agent/manager/sync.go#L78

And  there is not  previous locks there, we have locks when writting and in this getter when searching, 
the delete on that line must never be called, if it is called it is a bug.
So adding write lock there is not `neccessary`

Reply:
If we're going to delete the stale entries, we absolutely do have to hold the write lock. Either we don't bother deleting the stale entries because we're sure that it won't ever be required, or we hold the write lock.

Reply:
ok, so, are both ok with update that lock to ULock?

Reply:
Yes, holding the write lock SGTM.

Reply:
done